modules/application/src/main/java/edu/northeastern/truthtree/adapter/basicInfo/BasicInfoDBAdapter.java

```diff
@@ -148,22 +156,81 @@ public JSONArray getBasicCitiesInfo(int startValue, int endValue) {
    * @return JSONArray that contains states that are within the provided range.
    */
   @Override
-  public JSONArray getBasicCountiesInfo(int startValue, int endValue) {
-    JSONArray response = getPopulationData(startValue, endValue, 1);
-    return addPopulationToBasicInfo(countiesData, response);
+  public PageDTO<LocationDTO> getBasicCountiesInfo(int page, int startValue, int endValue)
+          throws IOException {
+    return getPopulationData(page, startValue, endValue, countiesData, 1);
+  }
+
+  private PageDTO<LocationDTO> getPopulationData(int currentPage, int startValue, int endValue,
+                                 Map<Long, Object> basicInfoMap, int typeCode)
+          throws IOException {
+    String url = UriComponentsBuilder
+            .fromHttpUrl(POPULATION_RANGE_URL)
+            .queryParam("attributeId", POPULATION_ID)
+            .queryParam("year", 2016)
+            .queryParam("from", startValue)
+            .queryParam("to", endValue)
+            .queryParam("currentPage", currentPage)
+            .queryParam("typeCode", typeCode)
+            .queryParam("pageSize", 50)
+            .toUriString();
+
+    ObjectMapper mapper = new ObjectMapper();
+    ResponseEntity<String> response = restTemplate.getForEntity(url, String.class);
+    Map<String, Object> bodyMap = mapper.readValue(response.getBody(),
+            new TypeReference<HashMap<String, Object>>(){});
+    return mapToLocationPage((Map) bodyMap.get("data"), basicInfoMap);
+  }
+
+  private PageDTO<LocationDTO> mapToLocationPage(Map<String, Object> dataMap,
+                                                 Map<Long, Object> basicInfoMap) {
+    List<Map<String, Object>> populationMaps = (List) dataMap.get("data");
+    return new PageDTO
+            .Builder()
+            .withCurrentPage((int) dataMap.get("currentPage"))
+            .withTotalPageCount((int) dataMap.get("totalPage"))
+            .withTotalItemCount((int) dataMap.get("total"))
+            .withItems(populationMaps
+                    .stream()
+                    .map(populationMap -> this.mapToLocationDTO(populationMap, basicInfoMap))
+                    .collect(Collectors.toList()))
+            .build();
+  }
+
+  private LocationDTO mapToLocationDTO(Map<String, Object> populationMap,
```

Comment:
This method can be added to assembler class.

Reply:
addressed